data/slash/multipleHeaderSigningProof.go

```diff
@@ -1,2 +1,110 @@
 //go:generate protoc -I=. -I=$GOPATH/src/github.com/ElrondNetwork/elrond-go-core/data/block -I=$GOPATH/src -I=$GOPATH/src/github.com/ElrondNetwork/protobuf/protobuf --gogoslick_out=. multipleHeaderSigningProof.proto
 package slash
+
+import (
+	"sort"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/data"
+)
+
+// GetType returns MultipleSigning
+func (m *MultipleHeaderSigningProof) GetType() SlashingType {
+	if m == nil {
+		return None
+	}
+	return MultipleSigning
+}
+
+// GetLevel returns the ThreatLevel of a possible malicious validator
+func (m *MultipleHeaderSigningProof) GetLevel(pubKey []byte) ThreatLevel {
+	if m == nil {
+		return Zero
+	}
+
+	level, exists := m.Levels[string(pubKey)]
+	if !exists {
+		return Zero
+	}
+
+	return level
+}
+
+// GetHeaders returns all headers that have been signed by a possible malicious validator
+func (m *MultipleHeaderSigningProof) GetHeaders(pubKey []byte) []data.HeaderHandler {
+	if m == nil {
+		return nil
+	}
+
+	headersV2, exist := m.HeadersV2[string(pubKey)]
+	if !exist {
+		return nil
+	}
+
+	return headersV2.GetHeaderHandlers()
+}
+
+// GetProofTxData returns the necessary ProofTxData to issue a commitment slash tx
+func (m *MultipleHeaderSigningProof) GetProofTxData() (*ProofTxData, error) {
+	if m == nil {
+		return nil, data.ErrNilPointerReceiver
+	}
+
+	pubKeys := m.GetPubKeys()
+	if len(pubKeys) == 0 {
+		return nil, data.ErrNotEnoughPublicKeysProvided
+	}
+	pubKey := pubKeys[0]
+	headers := m.GetHeaders(pubKey)
+	if len(headers) == 0 {
```

Comment:
Because the proof struct should not hold the "validity logic" inside and check for all headers to have the same round. This is the job of the `detector` which offers a func to `ValidateProof`.
The way I see it, this proof struct is a convenient holder for proof data and should not perform validity checks. 
If someone runs an "altered code" of the `detector`, which could create wrong multiple header singing proofs(for instance, here, containing headers with different rounds), then issues an invalid proof commitment tx, then he shall be penalized accordingly. For every other "non-altered" code, checking the round only for one random header should be enough.

However, if you consider that we should change this, let me know and we can discuss and eventually change it.

Reply:
right, I didn't check what you return here, so you only needed the round and shardID, then it should be fine.